packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -87,38 +82,71 @@ class AbstractModal extends React.Component {
   constructor() {
     super();
     this.handleKeydown = this.handleKeydown.bind(this);
+    this.setAbstractModalRef = this.setAbstractModalRef.bind(this);
+    this.showModalDomUpdates = this.showModalDomUpdates.bind(this);
+    this.hideModalDomUpdates = this.hideModalDomUpdates.bind(this);
   }
 
   componentDidMount() {
     document.addEventListener('keydown', this.handleKeydown);
+
+    if (this.props.isOpen) {
+      this.showModalDomUpdates();
+    }
   }
 
   componentDidUpdate(prevProps) {
-    // When the Modal is no longer in focus, it should no longer listen to the keydown event to handle the Escape key.
-    // Otherwise, the Modal would also get closed when the intention for pressing the Escape key is to close a popup inside the modal.
-    if (!this.props.isFocused && prevProps.isFocused) {
-      document.removeEventListener('keydown', this.handleKeydown);
-    } else if (this.props.isFocused && !prevProps.isFocused) {
-      document.addEventListener('keydown', this.handleKeydown);
+    if (this.props.isOpen) {
+      if (!prevProps.isOpen) {
+        this.showModalDomUpdates();
+      }
+    } else if (prevProps.isOpen) {
+      this.hideModalDomUpdates();
     }
   }
 
   componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeydown);
-    this.toggleVisuallyHiddenMainDocument('false');
+    this.hideModalDomUpdates();
   }
 
-  handleKeydown(e) {
-    if (e.keyCode === KeyCode.KEY_ESCAPE && this.props.isOpen && this.props.closeOnEsc && this.props.isFocused) {
-      this.props.onRequestClose();
+  setAbstractModalRef(node) {
+    this.AbstractModalRef = node;
+  }
+
+  showModalDomUpdates() {
+    const mainDocumentElement = document.querySelector(this.props.rootSelector);
+
+    if (mainDocumentElement) {
+      mainDocumentElement.setAttribute('aria-hidden', 'true');
+      mainDocumentElement.setAttribute('inert', '');
```

Comment:
should inert be set to true?

Reply:
Nah, `inert` is similar to the `disabled` HTML attribute where its presence equals it being set to true. https://github.com/WICG/inert#2-toggle-inert-on-an-element